assets/src/components/imageUpload/index.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Image uploader component.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { data } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import './style.scss';
+
+/**
+ * Image select/upload button and modal.
+ */
+class ImageUpload extends Component {
+
+	/**
+	 * Constructor.
+	 */
+	constructor( props ) {
```

Comment:
Tiny nit, but we could probably remove `props` from the function signature since it's not being used anymore.